File: src/fit/ScientificDoubleTest.java
// Modified or written by Object Mentor, Inc. for inclusion with FitNesse.
// Copyright (c) 2002 Cunningham & Cunningham, Inc.
// Released under the terms of the GNU General Public License version 2 or later.
package fit;

import junit.framework.TestCase;

public class ScientificDoubleTest extends TestCase {
  public void testScientificDouble() {
    Double pi = new Double(3.141592653589793);
    assertEquals(ScientificDouble.valueOf("3.14"), pi);
    assertEquals(ScientificDouble.valueOf("3.142"), pi);
    assertEquals(ScientificDouble.valueOf("3.1416"), pi);
    assertEquals(ScientificDouble.valueOf("3.14159"), pi);
    assertEquals(ScientificDouble.valueOf("3.141592653589793"), pi);
    assertFalse(ScientificDouble.valueOf("3.140").equals(pi));
    assertFalse(ScientificDouble.valueOf("3.144").equals(pi));
    assertFalse(ScientificDouble.valueOf("3.1414").equals(pi));
    assertFalse(ScientificDouble.valueOf("3.141592863").equals(pi));
    assertEquals(ScientificDouble.valueOf("6.02e23"), new Double(6.02e23));
    assertEquals(ScientificDouble.valueOf("6.02E23"), new Double(6.024E23));
    assertEquals(ScientificDouble.valueOf("6.02e23"), new Double(6.016e23));
    assertFalse(ScientificDouble.valueOf("6.02e23").equals(new Double(6.026e23)));
    assertFalse(ScientificDouble.valueOf("6.02e23").equals(new Double(6.014e23)));
  }

}

File: dotnet/src/fit/ScientificDoubleTest.cs
// Copyright (C) 2003,2004 by Object Mentor, Inc. All rights reserved.
// Released under the terms of the GNU General Public License version 2 or later.
using NUnit.Framework;

namespace fit
{
	[TestFixture]
	public class ScientificDoubleTest
	{
		[Test]
		public void TestScientificDouble()
		{
			double pi = 3.141592865;
			Assert.AreEqual(ScientificDouble.ValueOf("3.14"), pi);
			Assert.AreEqual(ScientificDouble.ValueOf("3.142"), pi);
			Assert.AreEqual(ScientificDouble.ValueOf("3.1416"), pi);
			Assert.AreEqual(ScientificDouble.ValueOf("3.14159"), pi);
			Assert.AreEqual(ScientificDouble.ValueOf("3.141592865"), pi);
			Assert.IsTrue(!ScientificDouble.ValueOf("3.140").Equals(pi));
			Assert.IsTrue(!ScientificDouble.ValueOf("3.144").Equals(pi));
			Assert.IsTrue(!ScientificDouble.ValueOf("3.1414").Equals(pi));
			Assert.IsTrue(!ScientificDouble.ValueOf("3.141592863").Equals(pi));
			Assert.AreEqual(ScientificDouble.ValueOf("6.02e23"), 6.02e23d);
			Assert.AreEqual(ScientificDouble.ValueOf("6.02E23"), 6.024E23d);
			Assert.AreEqual(ScientificDouble.ValueOf("6.02e23"), 6.016e23d);
			Assert.IsTrue(!ScientificDouble.ValueOf("6.02e23").Equals(6.026e23d));
			Assert.IsTrue(!ScientificDouble.ValueOf("6.02e23").Equals(6.014e23d));
			Assert.AreEqual(ScientificDouble.ValueOf("3.14"), ScientificDouble.ValueOf("3.14"));
		}
	}
}
